File: all-submissions/0907-koko-eating-bananas/solution.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int minSpeed = 1;
        int maxSpeed = Integer.MIN_VALUE;
        for (int pile : piles) {
            maxSpeed = Math.max(maxSpeed, pile);
        }
        while(minSpeed<maxSpeed) {
            int midSpeed = (minSpeed+maxSpeed)/2;
            int hoursSpent = 0;
            for(int pile: piles) {
                hoursSpent += Math.ceil((double) pile/midSpeed);
            }
            if(hoursSpent > h){
                minSpeed = midSpeed +1;
            } 
            else {
                maxSpeed = midSpeed;
            }
        }
        return minSpeed;
    }
}

File: all-submissions/0907-koko-eating-bananas/solution.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        def check(val):
            hours = 0
            for bananas in piles:
                hours += ceil(bananas/val)
            return hours <= h
        left = 1
        right = max(piles)
        while(left <= right):
            mid = (left + right) // 2
            if check(mid):
                right = mid-1
            else:
                left = mid+1
        return left
